plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -99,7 +100,19 @@ public FlightClientManager(
      * @return An OpenSearchFlightClient instance for the specified node
      */
     public Optional<FlightClient> getFlightClient(String nodeId) {
-        return Optional.ofNullable(flightClients.get(nodeId));
+        ClientHolder clientHolder = flightClients.getOrDefault(nodeId, null);
```

Comment:
You will always get `null` when key is not in the map:

```suggestion
        ClientHolder clientHolder = flightClients.get(nodeId);
```